### ISSUE TITLE
this string is expanded with {} preserved

### ISSUE THREAD

**Issue by iomintz** [facelessuser/bracex#14]
```py
🐍 import bracex
🐍 bracex.expand('{cautious_memory.{cogs.{permissions,wiki,watch_lists}.{db,commands},api,meta},jishaku,bot_bin.{misc,debug,sql,stats}}')
['cautious_memory.{cogs.permissions.db}', 'cautious_memory.{cogs.permissions.commands}', 'cautious_memory.{cogs.wiki.db}', 'cautious_memory.{cogs.wiki.commands}', 'cautious_memory.{cogs.watch_lists.db}', 'cautious_memory.{cogs.watch_lists.commands}', 'cautious_memory.{cogs.api}', 'cautious_memory.{cogs.meta}', 'jishaku', 'bot_bin.misc', 'bot_bin.debug', 'bot_bin.sql', 'bot_bin.stats']
🐍 bracex.__version__
'1.3'
```
## Expected
That, but without the {} in the return value

**Comment by iomintz** [comment]
Closing. `{cautious_memory.cogs.{{permissions,wiki,watch_lists}.{db,commands},api,meta},jishaku,bot_bin.{misc,debug,sql,stats}}` is the correct way to do this

**Comment by facelessuser** [comment]
Glad you got it figured out. I was just taking a look, now I don't have too 🙂 .

**Comment by facelessuser** [comment]
Interestingly, this is what I got when running your original example and new example:

```py
>>> bracex.expand('{cautious_memory.{cogs.{permissions,wiki,watch_lists}.{db,commands},api,meta},jishaku,bot_bin.{misc,debug,sql,stats}}')
['cautious_memory.cogs.permissions.db', 'cautious_memory.cogs.permissions.commands', 'cautious_memory.cogs.wiki.db', 'cautious_memory.cogs.wiki.commands', 'cautious_memory.cogs.watch_lists.db', 'cautious_memory.cogs.watch_lists.commands', 'cautious_memory.api', 'cautious_memory.meta', 'jishaku', 'bot_bin.misc', 'bot_bin.debug', 'bot_bin.sql', 'bot_bin.stats']
bracex.expand('{cautious_memory.cogs.{{permissions,wiki,watch_lists}.{db,commands},api,meta},jishaku,bot_bin.{misc,debug,sql,stats}}')
['cautious_memory.cogs.permissions.db', 'cautious_memory.cogs.permissions.commands', 'cautious_memory.cogs.wiki.db', 'cautious_memory.cogs.wiki.commands', 'cautious_memory.cogs.watch_lists.db', 'cautious_memory.cogs.watch_lists.commands', 'cautious_memory.cogs.api', 'cautious_memory.cogs.meta', 'jishaku', 'bot_bin.misc', 'bot_bin.debug', 'bot_bin.sql', 'bot_bin.stats']
>>> bracex.__version__
'1.3'
```

I did not see braces in the return. Was this something you added to illustrate your point?  I was concerned that braces where not being handled proper, and I was comparing to bash, but then I see I didn't get what you did. I am assuming you added the braces?

**Comment by iomintz** [comment]
I did get the braces, somehow… https://github.com/facelessuser/bracex/blob/1.3.0/bracex/__init__.py#L262-L268

**Comment by facelessuser** [comment]
Hmm, with 1.3, I don't get those braces as you show in your example. I literally copied and pasted your example and got no braces.

Bracex mimics bash logic as close as possible. The lines you pointed to have a comment specifically mentioning different handling of top level braces, which does not include nested braces. This too mimics bash. We actually run cases through bash and compare our implementation to bash's results.

Based on my tests, the opening example in this issue is not reproducible. Maybe a mistake in copy and paste? Maybe you were trying different things and copied the wrong example?

**Comment by facelessuser** [comment]
Yeah, I've tested out the opening example with a fresh install from pip on macOS, Linux, and Windows, and none of them returned braces in the results. I am confident, that if the following is returning braces, it is not the implementation I am shipping in 1.3. Maybe you were comparing the different examples to someone else's implementation and confused the two?  Or somehow, my implementation was modified locally?

Bracex doesn't really have platform specific code, so I was expecting a difference across OS types, but I checked to be sure.

I am more than happy to look into any issues that exhibit a behavior that differs from bash's that is reproducible. In any event, I suspect whatever caused the brace problem for you is somehow not what I am shipping.

I am open to be proven wrong, and If this is truly something that is happening, I am more than happy to fix it, but I am struggling to reproduce.
